apps/shipments/serializers.py

```diff
@@ -115,9 +116,7 @@ def create(self, validated_data):
             for location_field in ['ship_from_location', 'ship_to_location', 'bill_to_location']:
                 if location_field in validated_data:
                     data = validated_data.pop(location_field)
-                    if 'owner_id' not in data:
-                        data['owner_id'] = validated_data['owner_id']
-                    extra_args[location_field], _ = Location.objects.get_or_create(**data)
+                    extra_args[location_field] = Location.objects.create(**data)
```

Comment:
Shouldn't we include the `final_destination_location` in the for loop list?

Reply:
The reason why I left it out is because it that `final_destination_location` should only really be set when `delivery_act` is set at the end of a shipment. I don't think it makes sense to set during shipment creation because it would always be the same as `ship_to_location`. But I'm not sure. What do you think?

Reply:
That totally makes sense. :+1: